tools/cldr-code/src/main/java/org/unicode/cldr/test/DisplayAndInputProcessor.java

```diff
@@ -1218,4 +1215,31 @@ private static boolean wantsNNBSP(String path) {
             return false;
         }
     }
+
+    private static final Pattern ZERO_WIDTH_SPACES = PatternCache.get("\\u200B+");
+    private static final Set<String> LOCALES_NOT_ALLOWING_ZWS = new HashSet<>(Arrays.asList("da", "fr"));
+
+    /**
+     * Remove occurrences of U+200B ZERO_WIDTH_SPACE under certain conditions
+     *
+     * @param value the value to be normalized
+     * @return the normalized value
+     *
+     * TODO: extend this method to address more concerns, after clarifying the conditions
+     *   - enlarge the set LOCALES_NOT_ALLOWING_ZWS?
+     *   - strip initial and final ZWS in all locales?
+     *   - reduce two or more adjacent ZWS to one ZWS?
+     *   - allow or prohibit ZWS by itself as currency symbol, as currently in locales kea, pt_CV, pt_PT
+     *   - allow or prohibit ZWS preceding URL as in "as per [U+200B]http://unicode.org/repos/cldr/trunk/specs/ldml/tr35-general.html#Annotations"
+     * Reference: https://unicode-org.atlassian.net/browse/CLDR-15976
+     */
+    private String normalizeZeroWidthSpace(String value) {
+        if (ZERO_WIDTH_SPACES.matcher(value).find()) {
+            final String localeId = locale.getBaseName();
+            if (LOCALES_NOT_ALLOWING_ZWS.contains(localeId)) {
```

Comment:
This excludes fixing fr-CA etc., right? Maybe check only the language subtag, or language+script?

Reply:
That's a good idea for how this code could be improved in v43. As it stands for last-minute correction in v42, there's no actual occurrence of ZWS in fr_CA or other sublocales of "fr" or "da". For v43, I suggest we investigate possible mechanisms for getting a much much larger more comprehensive set of LOCALES_NOT_ALLOWING_ZWS. Might there be cases where variants of the same language differ in whether they should be included in the set?

Reply:
@markusicu  true, but da and fr are not the only locales that should exclude this.  I think the algorithm is to be improved in the future.


No other main/*.xml files had unexpected zwsp, except root.xml has it in the garbled part of the top comment block (!!)

Reply:
@btangmu 

> Might there be cases where variants of the same language differ in whether they should be included in the set?

Does not seem likely.